# Build - in - functions

## 1. print


In [21]:
print("hello vishal")

hello vishal


## 2. input

In [22]:
# input("Enter your name : ")

## 3. type

In [23]:
a = True  #4.5 or 75 or 3.333 or H 

type(a)

bool

## 4. int etc

In [46]:
int(5.6)
# float, string , list , truple

5

## 5. abs (Absolute)

In [25]:
abs(-4)

4

## 6. pow  (power)

In [26]:
pow(2,3) #2**3 , 3 is power of 2

8

## 7. min/max

#### The min() function returns the smallest value.
#### The max() function returns the largest value.

In [27]:
min([2,4,21,5,8])

#max([21,24,31,4,8])  output is 31

2

In [28]:
print(min("apple", "banana", "cherry"))
print(max("apple", "banana", "cherry"))



apple
cherry


In [29]:
print(min("python"))
print(max("python"))

h
y


In [30]:
fruits = ["mango", "apple", "banana"]

print(min(fruits))
print(max(fruits))

apple
mango


## 8. round

#### The round() function is used to round a number to the nearest integer or a specified number of decimal places.

In [ ]:
print(round(4.6)) #after 0.5 its count in next number 
print(round(4.2))
print(round(3.14159, 2)) # here 2 is define number of digits after decimal you want.print(round(1234, -1))
print(round(1234, -2))

5
4
3.14
1200


## 9. divmod 

#### The divmod() function returns both quotient and remainder when one number is divided by another.

In [ ]:
# Instead of writing:
# quotient = 17 // 5 => 3
# remainder = 17 % 5 => 2

# you can use:
# divmod(quotient, remainder)

# divmod(17, 5)

q, r = divmod(17, 5)

print("Quotient =", q)
print("Remainder =", r)

Quotient = 3
Remainder = 2


In [37]:
friends = 5
chocolates = 23

q, r = divmod(chocolates, friends)

print("Each friend gets:", q)
print("Leftover chocolates:", r)

Each friend gets: 4
Leftover chocolates: 3


## 10. id

#### The id() function returns the unique identity (memory address) of an object.

In [38]:
x = 10

print(id(x))

140713086928280


## 11. bin/oct /hext

#### These functions are used to convert a decimal (base 10) number into:
<!-- 
bin() → Binary (base 2)
oct() → Octal (base 8)
hex() → Hexadecimal (base 16) -->

In [43]:
num = 25

print("Binary:", bin(num))
print("Octal:", oct(num))
print("Hexadecimal:", hex(num))

Binary: 0b11001
Octal: 0o31
Hexadecimal: 0x19


## 12. ord

#### The ord() function returns the Unicode (ASCII) value of a character.

In [44]:
ord('C')

67

## 13 . len

#### The len() function returns the number of items in an object.


In [ ]:
name = "Vishal"

print(len(name))

# V i s h a l
# 1 2 3 4 5 6

numbers = [10, 20, 30, 40]

print(len(numbers))
# 4

# 10 , 20 , 30 ,40 
# 1  , 2,   3,  4   they count the elements of objects 

6
4


## 14 . sum

#### The sum() function returns the total of all numbers in an iterable.

In [ ]:
numbers = [10, 20, 30, 40]

print(sum(numbers))

# sum of all number 

100


## 15. help 


#### The help() function is used to get information, documentation, and details about Python functions, modules, classes, and keywords.

In [49]:
help(print)
help(len)
help(str)

Help on built-in function print in module builtins:

print(*args, sep=' ', end='\n', file=None, flush=False)
    Prints the values to a stream, or to sys.stdout by default.

    sep
      string inserted between values, default a space.
    end
      string appended after the last value, default a newline.
    file
      a file-like object (stream); defaults to the current sys.stdout.
    flush
      whether to forcibly flush the stream.

Help on built-in function len in module builtins:

len(obj, /)
    Return the number of items in a container.

Help on class str in module builtins:

class str(object)
 |  str(object='') -> str
 |  str(bytes_or_buffer[, encoding[, errors]]) -> str
 |
 |  Create a new string object from the given object. If encoding or
 |  errors is specified, then the object must expose a data buffer
 |  that will be decoded using the given encoding and error handler.
 |  Otherwise, returns the result of object.__str__() (if defined)
 |  or repr(object).
 |  encoding 

<!--| Function   | Example                     |
| ---------- | --------------------------- |
| `abs()`    | `abs(-25)` → `25`           |
| `min()`    | `min(10,5,3)` → `3`         |
| `max()`    | `max(10,5,3)` → `10`        |
| `round()`  | `round(3.14159,2)` → `3.14` |
| `divmod()` | `divmod(17,5)` → `(3,2)`    |
| `id()`     | `id(x)` → memory identity   |
| `bin()`    | `bin(10)` → `0b1010`        |
| `oct()`    | `oct(10)` → `0o12`          |
| `hex()`    | `hex(10)` → `0xa`           |
| `ord()`    | `ord('A')` → `65`           |
| `len()`    | `len("Python")` → `6`       |
| `sum()`    | `sum([1,2,3])` → `6`        |
| `help()`   | `help(print)`               |
| `chr()`    | `chr(65)` → `'A'`           |
  -->

 | Function   | Example                     |
| ---------- | --------------------------- |
| `abs()`    | `abs(-25)` → `25`           |
| `min()`    | `min(10,5,3)` → `3`         |
| `max()`    | `max(10,5,3)` → `10`        |
| `round()`  | `round(3.14159,2)` → `3.14` |
| `divmod()` | `divmod(17,5)` → `(3,2)`    |
| `id()`     | `id(x)` → memory identity   |
| `bin()`    | `bin(10)` → `0b1010`        |
| `oct()`    | `oct(10)` → `0o12`          |
| `hex()`    | `hex(10)` → `0xa`           |
| `ord()`    | `ord('A')` → `65`           |
| `len()`    | `len("Python")` → `6`       |
| `sum()`    | `sum([1,2,3])` → `6`        |
| `help()`   | `help(print)`               |
| `chr()`    | `chr(65)` → `'A'`           |
 